### PR TITLE
Rewrite ListActions component

### PR DESCRIPTION
**Pull request by parostatkiem**
**Description**

Changes proposed in this pull request:

- add ability to display some actions in a dropdown, some separately
- add ability to display icons for actions that are not in the dropdown
- set predefined icons for actions named `Edit` or `Delete`

**Related issue(s)**

### CODE REVIEW

File: shared/components/GenericList/test/GenericList.test.js [hunk 1]
```diff
@@ -1,86 +1,231 @@
...
+    const { queryByText } = render(
+      <GenericList
+        entries={[mockEntries[0]]}
+        headerRenderer={mockHeaderRenderer}
+        rowRenderer={customEntryRenderer}
+      />,
+    );
+
+    expect(await queryByText(mockEntries[0].name)).toBeTruthy();
+    // expect(await queryByText('maskopatol')).toBeTruthy();
```
Comment (akucharska) [inline shared/components/GenericList/test/GenericList.test.js]: Delete :)

File: shared/components/GenericList/test/GenericList.test.js [hunk 2]
```diff
@@ -1,86 +1,231 @@
...
     );
-    let tree = component.toJSON();
-    expect(tree).toMatchSnapshot();
+    await getByText(defaultNotFoundText);
   });
 
-  it('Renders title and description', () => {
-    const [title, description] = ['title', 'description'];
-    const component = renderer.create(
+  it('Renders custon notFoundMessage props', async () => {
```
Comment (akucharska) [inline shared/components/GenericList/test/GenericList.test.js]: Misspelling 'custom'

File: shared/components/GenericList/test/GenericList.test.js [hunk 3]
```diff
@@ -1,86 +1,231 @@
...
+          headerRenderer={mockHeaderRenderer}
+          rowRenderer={mockEntryRenderer}
+          showSearchField={false}
+        />,
+      );
+
+      expect(await queryByRole('search')).toBeNull();
+    });
+
+    it("Desn't show search field when showSearchField is set to false", async () => {
```
Comment (akucharska) [inline shared/components/GenericList/test/GenericList.test.js]: Duplicated test

File: shared/components/GenericList/test/GenericList.test.js [hunk 4]
```diff
@@ -1,86 +1,231 @@
...
+      );
+
+      expect(await queryAllByRole('row')).toHaveLength(3); // header + {mockEntries.length} rows
+
+      const searchInput = await getByLabelText('search-input');
+      fireEvent.change(searchInput, { target: { value: searchText } });
+
+      expect(await queryAllByRole('row')).toHaveLength(2); // header + NotFoundMessage dedicated row
+      await getByText(notFoundMessage);
+    });
```
Comment (akucharska) [inline shared/components/GenericList/test/GenericList.test.js]: Add test scenario:
it("Doesn't render empty cell in table header when there is no actions", async () => {
});

Also make sure <th> is not rendered when there is no actions

**Comment by parostatkiem** [comment]
@akucharska ✅

**Pull request merged**